lib/auth/auth.js

```diff
@@ -26,120 +30,151 @@ auth.setAuthHandler = handler => {
  * @param {string} awsService - Aws service related
  * @param {object} data - Parameters from queryString parsing or body of
  *      POST request
- * @return {object} Return object with information to authenticate
+ *
+ * @return {object} ret
+ * @return {object} ret.err - arsenal.errors object if any error was found
+ * @return {object} ret.params - auth parameters to use later on for signature
+ *                               computation and check
+ * @return {object} ret.params.version - the auth scheme version
+ *                                       (undefined, 2, 4)
+ * @return {object} ret.params.data - the auth scheme's specific data
  */
-auth.check = (request, log, awsService, data) => {
-    log.debug('running auth checks', { method: 'auth' });
+function extractParams(request, log, awsService, data) {
+    log.trace('extracting authentication signature informations');
```

Comment:
informations s/b information
do we want to use the "method" log format instead and state the function name?